x-pack/plugins/security_solution/common/endpoint/generate_data.ts

```diff
@@ -75,21 +75,98 @@ const POLICIES: Array<{ name: string; id: string }> = [
 const FILE_OPERATIONS: string[] = ['creation', 'open', 'rename', 'execution', 'deletion'];
 
 interface EventInfo {
-  category: string;
+  category: string | string[];
   /**
    * This denotes the `event.type` field for when an event is created, this can be `start` or `creation`
    */
   creationType: string;
 }
 
+/**
+ * The valid ecs categories.
+ */
+export enum ECSCategory {
+  Driver = 'driver',
+  File = 'file',
+  Network = 'network',
+  /**
+   * Registry has not been added to ecs yet.
+   */
+  Registry = 'registry',
+  Authentication = 'authentication',
+  Session = 'session',
+}
+
+/**
+ * High level categories for related events. These specify the type of related events that should be generated.
+ */
+export enum RelatedEventCategory {
+  /**
+   * The Random category allows the related event categories to be chosen randomly
+   */
+  Random = 'random',
+  Driver = 'driver',
+  File = 'file',
+  Network = 'network',
+  Registry = 'registry',
+  /**
+   * Security isn't an actual category but defines a type of related event to be created.
+   */
+  Security = 'security',
+}
+
+/**
+ * This map defines the relationship between a higher level event type defined by the RelatedEventCategory enums and
+ * the ECS categories that is should map to. This should only be used for tests that need to determine the exact
+ * ecs categories that were created based on the related event information passed to the generator.
+ */
+export const categoryMapping: Record<RelatedEventCategory, ECSCategory | ECSCategory[] | ''> = {
+  [RelatedEventCategory.Security]: [ECSCategory.Authentication, ECSCategory.Session],
```

Comment:
I'm not sure we should create a higher level category like this, this kind of logic would have to be duplicated anywhere a user tries to interact with the data, e.g. kql/eql, and might be more effort than it's worth

Reply:
This isn't actually emitting non-ECS categories right? It's just metaprogramming logic used in the generator

Reply:
> I'm not sure we should create a higher level category like this, this kind of logic would have to be duplicated anywhere a user tries to interact with the data, e.g. kql/eql, and might be more effort than it's worth

So all this is trying to provide is a way for tests to say: I want to construct 4 related events and I want them have these particular types. It should only be used for tests.

The problem I ran into was that if you want the generator to produce an event that models what the endpoint would send for say a "security" event we need someway to tell the generator to build a security event which has two ecs categories. That's why I built the higher level "types".

> This isn't actually emitting non-ECS categories right? It's just metaprogramming logic used in the generator

That's correct.